### PR TITLE
Feature/prime 1770 swapping signing auth

### PR DESCRIPTION
**Pull request by hogmann**
Tests are updated, and added a silly little s to tranfer

### CODE REVIEW

File: prime-angular-frontend/src/app/core/resources/organization-resource.service.ts [hunk 1]
```diff
@@ -276,6 +275,42 @@ export class OrganizationResource {
       );
   }
 
+  /**
+     * @description
+     * Get care setting codes that still require a signature on an oganization that is
+     * pending a tranfer
```
Comment (Paahn) [inline prime-angular-frontend/src/app/core/resources/organization-resource.service.ts:281]: spelling check

File: prime-dotnet-webapi/Controllers/OrganizationsController.cs [hunk 2]
```diff
@@ -372,6 +373,62 @@ public async Task<ActionResult> UpdateOrganizationAgreement(int organizationId,
...
+            if (organization == null)
+            {
+                return NotFound($"Organization not found with id {organizationId}");
+            }
+            if (!organization.SigningAuthority.PermissionsRecord().AccessableBy(User))
+            {
+                return Forbid();
+            }
+
+            var careSettingCodes = await _organizationService.GetCareSettingCodesForPendingTranferAsync(organizationId, organization.SigningAuthorityId);
```
Comment (Paahn) [inline prime-dotnet-webapi/Controllers/OrganizationsController.cs:399]: missing "s" in Transfer, probably should be GetCareSettingCodesForPendingTransferAsync

File: prime-dotnet-webapi/Controllers/OrganizationsController.cs [hunk 3]
```diff
@@ -372,6 +373,62 @@ public async Task<ActionResult> UpdateOrganizationAgreement(int organizationId,
...
+
+            return Ok(careSettingCodes);
+        }
+
+        // PUT: api/Organizations/5/finalize-transfer
+        /// <summary>
+        /// Clear Pending Tranfer Flag on an organization.
```
Comment (Paahn) [inline prime-dotnet-webapi/Controllers/OrganizationsController.cs:406]: transfer

File: prime-dotnet-webapi/Controllers/OrganizationsController.cs [hunk 4]
```diff
@@ -372,6 +373,62 @@ public async Task<ActionResult> UpdateOrganizationAgreement(int organizationId,
...
+            if (organization == null)
+            {
+                return NotFound($"Organization not found with id {organizationId}");
+            }
+            if (!organization.SigningAuthority.PermissionsRecord().AccessableBy(User))
+            {
+                return Forbid();
+            }
+
+            await _organizationService.FinalizeTranferAsync(organizationId);
```
Comment (Paahn) [inline prime-dotnet-webapi/Controllers/OrganizationsController.cs:427]: FinalizeTransferAsync

File: prime-dotnet-webapi/Services/OrganizationService.cs [hunk 5]
```diff
@@ -239,6 +239,31 @@ public async Task<Agreement> EnsureUpdatedOrgAgreementAsync(int organizationId,
             }
         }
 
+        public async Task<IEnumerable<CareSettingType>> GetCareSettingCodesForPendingTranferAsync(int organizationId, int signingAuthorityId)
```
Comment (Paahn) [inline prime-dotnet-webapi/Services/OrganizationService.cs:242]: missing "s" in Transfer, probably should be GetCareSettingCodesForPendingTransferAsync

File: prime-dotnet-webapi/Services/OrganizationService.cs [hunk 6]
```diff
@@ -239,6 +239,31 @@ public async Task<Agreement> EnsureUpdatedOrgAgreementAsync(int organizationId,
...
+
+            var agreementSettings = agreements
+                .GroupBy(a => a.AgreementVersion.AgreementType)
+                .Select(a => a.FirstOrDefault())
+                .Where(a => a.PartyId != signingAuthorityId || (a.PartyId == signingAuthorityId && a.AcceptedDate == null)).Select(a => SiteSettingForOrgAgreementType(a.AgreementVersion.AgreementType)).ToList();
+
+            return agreementSettings.Where(code => siteSettings.Contains((int)code)).ToList();
+        }
+
+        public async Task FinalizeTranferAsync(int organizationId)
```
Comment (Paahn) [inline prime-dotnet-webapi/Services/OrganizationService.cs:260]: FinalizeTransferAsync

File: prime-dotnet-webapi/Services/interfaces/IOrganizationService.cs [hunk 7]
```diff
@@ -21,10 +21,12 @@ public interface IOrganizationService
         Task<int> UpdateCompletedAsync(int organizationId);
         Task DeleteOrganizationAsync(int organizationId);
         Task<Organization> GetOrganizationNoTrackingAsync(int organizationId);
-        Task<Agreement> EnsureUpdatedOrgAgreementAsync(int organizationId, int siteId);
+        Task<Agreement> EnsureUpdatedOrgAgreementAsync(int organizationId, int careSettingCode, int signingAuthorityId);
         Task AcceptOrgAgreementAsync(int organizationId, int agreementId);
         Task<SignedAgreementDocument> AddSignedAgreementAsync(int organizationId, int agreementId, Guid documentGuid);
         Task<SignedAgreementDocument> GetLatestSignedAgreementAsync(int organizationId);
+        Task<IEnumerable<CareSettingType>> GetCareSettingCodesForPendingTranferAsync(int organizationId, int signingAuthorityId);
```
Comment (Paahn) [inline prime-dotnet-webapi/Services/interfaces/IOrganizationService.cs:28]: same missing "s" in Transfer

File: prime-dotnet-webapi/Services/interfaces/IOrganizationService.cs [hunk 8]
```diff
@@ -21,10 +21,12 @@ public interface IOrganizationService
...
+        Task FinalizeTranferAsync(int organizationId);
```
Comment (Paahn) [inline prime-dotnet-webapi/Services/interfaces/IOrganizationService.cs:29]: same as above

**Review by Paahn — CHANGES_REQUESTED** [review]

**Review by AMiKaH — COMMENTED** [review]
Other than the typos Panos mentioned, looks good.

**Review by Paahn — APPROVED** [review]
LGTM

File: prime-angular-frontend/src/app/modules/site-registration/pages/electronic-organization-agreement-page/electronic-organization-agreement-page.component.ts [hunk 9]
```diff
@@ -0,0 +1,101 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatCheckbox } from '@angular/material/checkbox';
+import { MatDialog } from '@angular/material/dialog';
+import { ActivatedRoute, Router } from '@angular/router';
+import { OrganizationResource } from '@core/resources/organization-resource.service';
+import { FormUtilsService } from '@core/services/form-utils.service';
+import { AbstractEnrolmentPage } from '@lib/classes/abstract-enrolment-page.class';
```
Comment (johnnyl6) [inline prime-angular-frontend/src/app/modules/site-registration/pages/electronic-organization-agreement-page/electronic-organization-agreement-page.component.ts:7]: unused import

**Review by johnnyl6 — APPROVED** [review]
LGTM

File: prime-angular-frontend/src/app/modules/site-registration/pages/electronic-organization-agreement-page/electronic-organization-agreement-page.component.ts (hunk 9 above)
Comment (hogmann) [reply prime-angular-frontend/src/app/modules/site-registration/pages/electronic-organization-agreement-page/electronic-organization-agreement-page.component.ts]: removed :)

**Pull request merged**